_posts/2021/2021-11-29-tidy-up-your-routes-with-only.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Tidy Up Your Routes with Only"
+description: "Be tidy"
+layout: article
+category: ruby
+image:
+  base: "2021/tidy-up-your-routes-with-only"
+  alt: "Road Closed. Diversion. Road signs."
+  credit: "Ben Wicks"
+  source: "https://unsplash.com/photos/zwN1MwCtR5Y"
+
+---
+
+It is easy with [Rails’ syntax for defining routes](https://guides.rubyonrails.org/routing.html) to open up more URLs in your application than you might intend.
+
+
+## Instead of…
+
+…using an open `resources` block in your routes:
+
+```ruby
+resources :orders do
+  resources :products
+end
+```
+
+## Use…
+
+…the `only` and `except` options to limit the actions you’re generating:
+
+```ruby
+resources :orders, except: %w[destroy] do
+  resources :products, only: %w[show]
+end
+```
+
+
+## Why?
+
+This is a case where it is all about clarity, tidying up and pretecting against unexpected errors or security holes.
+
+Primarily the issue is that in the default non-restricted case the seven default routes are created. If the correct path and method are passed into your application by a user, Rails will atempt to call the relevent controller action, even if it doesn’t exist, which will lead to errors.
```

Comment:
You're trying to say that if someone calls a route which is defined but not used, then an error is generated?

If so, that could be said more clearly!